src/System.Linq.Expressions/src/System/Runtime/CompilerServices/ReadOnlyCollectionBuilder.cs

```diff
@@ -418,9 +436,17 @@ public void Reverse()
         public void Reverse(int index, int count)
         {
             if (index < 0)
+            {
                 throw new ArgumentOutOfRangeException(nameof(index));
+            }
             if (count < 0)
+            {
                 throw new ArgumentOutOfRangeException(nameof(count));
+            }
+            if (Count - index < count)
+            {
+                throw new ArgumentException(SR.ArgumentInvalidOffLen);
+            }
```

Comment:
Yes I fixed next issue `Reverse lacks good bounds checks which can result in reversing a range of elements in the underlying array including the trailing empty slots, which then become visible`